Assignment4/MyLibrary/MyLibrary/NumSet.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MyLibrary
+{
+    public class NumSet
+    {
+        private HashSet<int>? mySet;
+
+        public HashSet<int>? MySet 
+        { 
+            get => mySet; 
+            set => mySet = value ??
+                throw new ArgumentNullException(nameof(value)); 
+        }
+
+        public NumSet(params int[] array)
+        {
+            mySet = new HashSet<int>(array);
+        }
+
+        public override string ToString()
+        {
+            return String.Join(", ", MySet!);
```

Comment:
```suggestion
            return string.Join(", ", MySet);
```
Use the C# keywords rather than BCL names.